targets/TARGET_STM/i2c_api.c

```diff
@@ -754,7 +755,8 @@ int i2c_read(i2c_t *obj, int address, char *data, int length, int stop) {
         timeout = BYTE_TIMEOUT_US * (length + 1);
         /*  transfer started : wait completion or timeout */
         while(!(obj_s->event & I2C_EVENT_ALL) && (--timeout != 0)) {
-            wait_us(1);
+            int start = us_ticker_read();
+            while ((us_ticker_read() - start) < 1);
```

Comment:
Few questions about this lines 
- us_ticker_read returns uint32_t - why using start as an int ? 
- what would be the resulting type of (us_ticker_read() - start)  and how to make sure that wrap-around case is well covered ? rather than < 1,maybe != 0 would be safer 
-  In any case,  with current proposal, the start will happen any time between T0 and T0+1, so when ((us_ticker_read() - start) < 1) is true, we have not reached a 1µs time, but rather any time between 0 and 1µs, in random way, so that the result will probably be around 0,5µs in average.  The actual measured timeout will be half the expected one. Most probably OK as we take margins in the timeout computation, nevertheless not what we expect to code.